src/pint/models/binary_ddk.py

```diff
@@ -98,7 +98,15 @@ def validate(self):
                     "DDK", "DDK model needs proper motion parameters."
                 )
         if self.SINI.quantity is not None:
-            raise ValueError(
+            raise TimingModelError(
                 "DDK model does not accept `SINI` as input. Please"
                 " use `KIN` instead."
             )
+
+        if hasattr(self._parent, "PX"):
+            if self._parent.PX.value == 0.0 or self._parent.PX.value is None:
```

Comment:
Is a parallax value of zero really invalid? Is an infinite distance not a valid value for this model? If you're going to be forbidding zero, should you also forbid negative values of PX?

Reply:
PX is zero is valid but not for DDK model. If you think the DDK model should accept zero, we should have another issue to discuss. This PR is designed to fix an internal bug and add some more checks. Having DDK accept zero PX requires a bigger discussion and more changes to the code.

Reply:
What about negative numbers?

Reply:
I changed the sign to <= 0.0. I am not sure why it does not show here.